tests/end-to-end/director/labeling_api_test.go

```diff
@@ -0,0 +1,845 @@
+
+	//WHEN
+	err = tc.RunQuery(ctx, createReq, &firstApp)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, firstApp.ID)
+	defer deleteApplication(t, firstApp.ID)
+
+	//Set label on application
+	// GIVEN
+	labelKey := "test"
+	labelValue := "val"
+
+	setLabelRequest := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+				result: setApplicationLabel(applicationID: %s, key: %s, value: %v) {
+						%s
+					}
+				}`,
+			firstApp.ID, labelKey, labelValue, tc.gqlFieldsProvider.ForApplicationLabel()))
+	label := graphql.Label{}
+
+	// WHEN
+	err = tc.RunQuery(ctx, setLabelRequest, &label)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, label.Key)
+	require.NotEmpty(t, label.Value)
+
+	// Check if LabelDefinition was created internally
+	// GIVEN
+	getLabelDefinitionRequest := gcli.NewRequest(
+		fmt.Sprintf(`query {
+				result: labelDefinition(key: %s) {
+						%s
+					}
+				}`,
+			"test", tc.gqlFieldsProvider.ForLabelDefinition()))
+	labelDefinition := graphql.LabelDefinition{}
+
+	// WHEN
+	err = tc.RunQuery(ctx, getLabelDefinitionRequest, &labelDefinition)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, labelDefinition)
+	assert.Equal(t, label.Key, labelDefinition.Key)
+}
+
+func TestCreateLabelAgainstLabelDefinition(t *testing.T) {
+	//Create application
+	// GIVEN
+	ctx := context.Background()
+
+	in := generateSampleApplicationInput("first")
+	appInputGQL, err := tc.graphqlizer.ApplicationInputToGQL(in)
+	require.NoError(t, err)
+
+	createReq := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+  				result: createApplication(in: %s) {
+    					%s
+					}
+				}`, appInputGQL, tc.gqlFieldsProvider.ForApplication()))
+	firstApp := graphql.Application{}
+
+	//WHEN
+	err = tc.RunQuery(ctx, createReq, &firstApp)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, firstApp.ID)
+	defer deleteApplication(t, firstApp.ID)
+
+	//Create LabelDefinition
+	// GIVEN
+	labelKey := "foo"
+	jsonSchema := `{
+	  "$id": "https://foo.com/bar.schema.json",
+	  "title": "foobarbaz",
+	  "type": "object",
+	  "properties": {
+		"foo": {
+		  "type": "string",
+		  "description": "foo"
+		},
+	  },
+      "required": ["foo"]
+}`
+
+	schema := interface{}(&jsonSchema)
+	labelDefinitionInput := graphql.LabelDefinitionInput{
+		Key:    labelKey,
+		Schema: &schema,
+	}
+
+	ldInputGql, err := tc.graphqlizer.LabelDefinitionInputToGQL(labelDefinitionInput)
+	require.NoError(t, err)
+
+	createLabelDefinitionRequest := gcli.NewRequest(
+		//    createLabelDefinition(in: LabelDefinitionInput!): LabelDefinition!
+		fmt.Sprintf(`mutation {
+				result: createLabelDefinition(in: %s) {
+						%s
+					}
+				}`,
+			ldInputGql, tc.gqlFieldsProvider.ForLabelDefinition()))
+	ld := graphql.LabelDefinition{}
+
+	// WHEN
+	err = tc.RunQuery(ctx, createLabelDefinitionRequest, ld)
+
+	// THEN
+	require.NoError(t, err)
+	assert.Equal(t, labelKey, ld.Key)
+
+	//Try to set label on application with invalid value against given schema
```

Comment:
I wonder if we should convert the comments to `t.Log` executions

Reply:
done